src/main/java/com/mabl/integration/jenkins/MablStepBuilder.java

```diff
@@ -365,27 +381,34 @@ private ListBoxModel getApplicationIdItems(Secret formApiKey, boolean disableSsl
                 }
 
                 return items;
-            } catch (IOException | MablSystemError e)  {
+            } catch (IOException | MablSystemException e)  {
                 LOGGER.warning("Failed to retrieve application IDs: " + e.getLocalizedMessage());
             }
 
             return getSelectValidApiKeyListBoxModel();
         }
 
-        public ListBoxModel doFillEnvironmentIdItems(@QueryParameter String restApiKeyId, @QueryParameter boolean disableSslVerification) {
+        public ListBoxModel doFillEnvironmentIdItems(
+                @QueryParameter String restApiKeyId,
+                @QueryParameter boolean disableSslVerification,
+                @QueryParameter String apiBaseUrl,
```

Comment:
Since this is a class, perhaps that URL only needs to be set once (e.g. on class init), so the method signatures don't need to change?